dev/integration_tests/flutter_gallery/lib/gallery/backdrop.dart

```diff
@@ -249,9 +249,11 @@ class _BackdropState extends State<Backdrop> with SingleTickerProviderStateMixin
   }
 
   void _toggleFrontLayer() {
-    final AnimationStatus status = _controller.status;
-    final bool isOpen = status == AnimationStatus.completed || status == AnimationStatus.forward;
-    _controller.fling(velocity: isOpen ? -2.0 : 2.0);
+    setState(() {
+      final AnimationStatus status = _controller.status;
+      final bool isOpen = status == AnimationStatus.completed || status == AnimationStatus.forward;
+      _controller.fling(velocity: isOpen ? -2.0 : 2.0);
+    });
```

Comment:
Why does this need a setState? There's no state changing in that callback.

Reply:
`_controller.fling` changes the state of the `_controller`, which `_buildStack` uses in a bunch of places. I haven't looked at which part exactly needs to be rebuilt so rebuilding the whole thing. It's possible that the rebuild can be scoped to one of the child widgets (maybe one of them fails to listen to the `AnimationController` state).

Reply:
Found it. The `builder` uses `_controller.status != AnimationStatus.completed`, so it needs a rebuild when the controller status changes.

Reply:
Alternatively I could add this in `initState`:

```dart
_controller.addStatusListener((AnimationStatus status) {setState(() {});});
```

Which one's more idiomatic?

Reply:
If the build is depending on the controller's status, the widget should listen to the controller's status and call setState when it changes.

Reply:
Done: https://github.com/flutter/flutter/pull/55414/commits/72d7d365be6639271a5e88ee3043b92833facb79